src/widgets/immunizations/immunizations-detailed-summary.component.tsx

```diff
@@ -0,0 +1,185 @@
+          immunizationFromConfig.vaccineUuid
+      );
+      if (matchingExistingImmunization) {
+        immunizationFromConfig.existingDoses =
+          matchingExistingImmunization.existingDoses;
+      }
+      return immunizationFromConfig;
+    });
+  };
+
+  useEffect(() => {
+    const abortController = new AbortController();
+    if (!isLoadingPatient && patient) {
+      const immunizationsConfig = props.immunizationsConfig;
+      const searchTerm = immunizationsConfig?.vaccinesConceptSet;
+      const configuredImmunizations: Promise<Array<
+        ImmunizationData
+      >> = getImmunizationsConceptSet(searchTerm, abortController).then(
+        findConfiguredSequences(immunizationsConfig?.sequenceDefinitions)
+      );
+
+      const existingImmunizationsForPatient: Promise<Array<
+        ImmunizationData
+      >> = performPatientImmunizationsSearch(
+        patient.identifier[0].value,
+        patientUuid,
+        abortController
+      ).then(mapFromFHIRImmunizationBundle);
+
+      const consolidatedImmunizations: Promise<Array<
+        ImmunizationData
+      >> = Promise.all([
+        configuredImmunizations,
+        existingImmunizationsForPatient
+      ]).then(([configuredImmunizations, existingImmunizationsForPatient]) =>
+        findExistingDoses(
+          configuredImmunizations,
+          existingImmunizationsForPatient
+        )
+      );
+
+      consolidatedImmunizations
+        .then((consolidatedImmunizations: Array<ImmunizationData>) => {
+          const sortedImmunizationsForPatient = orderBy(
+            consolidatedImmunizations,
+            [immunization => get(immunization, "existingDoses.length", 0)],
+            ["desc"]
+          );
+          setAllImmunizations(sortedImmunizationsForPatient);
+        })
+        .catch(err => {
+          if (err.name !== "AbortError") {
+            setAllImmunizations([]);
+            createErrorHandler();
+          }
+        });
+
+      return () => abortController.abort();
+    }
+  }, [isLoadingPatient, patient, patientUuid, props]);
+
+  function displayImmunizations() {
+    return (
+      <SummaryCard
+        name={t("Immunizations", "Immunizations")}
+        styles={{ width: "100%" }}
+      >
+        <table className={`omrs-type-body-regular ${styles.immunizationTable}`}>
+          <thead>
+            <tr>
+              <td>{t("vaccine", "VACCINE")}</td>
+              <td>{t("recent vaccination", "RECENT VACCINATION")}</td>
+              <td />
+            </tr>
+          </thead>
+          <tbody>
+            {allImmunizations &&
+              allImmunizations.map((immunizations, i) => {
+                return <VaccinationRow key={i} immunization={immunizations} />;
+              })}
+          </tbody>
+        </table>
+      </SummaryCard>
+    );
+  }
+
+  function displayNoImmunizations() {
+    return (
+      <SummaryCard
+        name={t("Immunizations", "Immunizations")}
+        styles={{
```

Comment:
Sure!

Reply:
@FlorianRappl 
I am not able to make this change! If I try to pass `classNames` to SummaryCard it is giving error `classNames` is not present on type `SummaryCardProps`.
I see in the whole app, summarycard sty;es are passed inline.

Reply:
I see - but this is rather a limitation of `SummaryCard`.

https://github.com/VasanthakumarTW/openmrs-esm-patient-chart-widgets/blob/Immunization-Widget/src/ui-components/cards/summary-card.component.tsx#L8

Just change the props to accept an optional `className` and pass this prop to the `className` computation:

```jsx
<div style={props.styles} className={`omrs-card ${styles.card} ${props.className || ''}`}>
```

Reply:
General remark here: This is why libs like emotion or styled components are better than CSS modules. We could do component inheritance here and override the style without having to fiddle with such details (and then potential problems like CSS specifity / declaration overwriting).

Reply:
Thanks @FlorianRappl!
Somehow I had the understanding since these are all components they should by default support `className` property and I don't need to explicitly define a prop for it. 
I thought classname is a react standard itself.

Reply:
No - there are only three standard props `children`, `key`, and `ref`. And the latter only appears when the component supports it - to support it in custom components you'd need to use something like `forwardRef`.

So we'd need to tell React where / how to use `className`.

Reply:
(Do you want to make this change in this PR or a dedicated one - I guess a dedicated one would make sense as it improves the whole patient-chart-widgets lib)

cc @mddubey

Reply:
Hello @FlorianRappl,
Sorry! I have actually already made this change in the [latest commit here](https://github.com/openmrs/openmrs-esm-patient-chart-widgets/pull/118/commits/b7e6c7275fc73f33dfbedbd0d27ff4963374906e). I have also made changes in the immunization components using it.

If you think this should be dedicated then we could revert last commit.